### ISSUE TITLE
Pointer to slice generates invalid code

### ISSUE THREAD

**Issue by jdmeyer3** [jmattheis/goverter#40]
**Describe the bug**
When a field of a struct contains a pointer to a slice, ie `*[]`, it generates invalid dereference code.

**To Reproduce**
 ```go
//go:generate go run github.com/jmattheis/goverter/cmd/goverter ...

package converter

// goverter:converter
type Converter interface {
	ConvertHouse(source DBHouseNames) APIHouseNames
}

type DBHouseNames struct {
	Names *[]string
}

type APIHouseNames struct {
	Names *[]string
}
 ```

This will end up generating code that contains invalid dereferences 
```go
...
*source.Names[i]
```

**Expected behavior**
it should generate code that properly dereferences the array before attempting to use the index
```go
(*source.Names)[i]
```

**Comment by jmattheis** [comment]
Good catch.

**Comment by jmattheis** [comment]
Fixed in v0.11.1.
